test/gtest/uct/test_md.cc

```diff
@@ -242,30 +244,44 @@ UCS_TEST_SKIP_COND_P(test_md, alloc,
                              UCT_MEM_ALLOC_PARAM_FIELD_NAME;
     params.flags           = UCT_MD_MEM_ACCESS_ALL;
     params.name            = "test";
-    params.mem_type        = UCS_MEMORY_TYPE_HOST;
     params.mds.mds         = &md_ref;
     params.mds.count       = 1;
 
-    for (unsigned i = 0; i < 300; ++i) {
-        size = orig_size = ucs::rand() % 65536;
-        if (size == 0) {
-            continue;
-        }
+    for (mem_type = UCS_MEMORY_TYPE_HOST; mem_type < UCS_MEMORY_TYPE_LAST; mem_type++) {
+        if (md_attr().cap.alloc_mem_types & UCS_BIT(mem_type)) {
+            if ((mem_type == UCS_MEMORY_TYPE_CUDA) ||
+                (mem_type == UCS_MEMORY_TYPE_CUDA_MANAGED)) {
+                if (!mem_type_ctx_created) {
+                    address = mem_buffer::allocate(1, (ucs_memory_type_t)mem_type);
+                    mem_buffer::release(address, (ucs_memory_type_t)mem_type);
```

Comment:
can we cudaSetDevice() to set context?  we can set context in test_md::init()?

Reply:
Is there a reason why want to directly call cuda API instead of mem_buffer functions? `allocate` and `release` will serve the same purpose right?

Agree with moving to init(). I'll make this change.